src/context-providers/builder/element.types.ts

```diff
@@ -40,10 +47,23 @@ export interface IPrefillAttributes {
     path?: string;
 }
 
+type MobileCol = 1 | 2 | 3 | 4;
+type MobileColRange = MobileCol | 5;
+type TabletCol = MobileCol | 5 | 6 | 7 | 8;
+type TabletColRange = TabletCol | 9;
+type DesktopCol = TabletCol | 9 | 10 | 11 | 12;
+type DesktopColRange = DesktopCol | 13;
+export interface IColumns {
+    desktop: DesktopCol | [DesktopColRange, DesktopColRange];
+    tablet: TabletCol | [TabletColRange, TabletColRange];
+    mobile: MobileCol | [MobileColRange, MobileColRange];
+}
+
 export interface IBaseFieldAttributes extends IBaseAttributes {
     label: string | undefined;
     required: boolean;
     requiredErrorMsg?: string | undefined;
+    columns?: IColumns;
```

Comment:
since we can be dealing with two-column layouts, it is prudent to ensure the columns are set all times

```suggestion
    columns: IColumns;
```